internal/factsengine/gatherers/saplocalhostresolver.go

```diff
@@ -0,0 +1,154 @@
+package gatherers
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"regexp"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/trento-project/agent/internal/core/sapsystem"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SapLocalhostResolverGathererName = "saplocalhost_resolver"
+)
+
+// nolint:gochecknoglobals
+var (
+	subgroupMapping = map[int]string{
```

Comment:
Could we squash this in the sprintf below? Wdyt? Completely up to you, We use is with len in other code, maybe a map is not the appropriate structure?